### PR TITLE
Make `maximum seconds that will be waited between reconnection attempts when connection is lost` to be configurable

### PR DESCRIPTION
**Pull request by moznion**
Now the default interval is `10m` that is configured at https://github.com/eclipse/paho.mqtt.golang/blob/9ec68b7894ac4cd48fc9126e910e571d50eb5d9c/options.go#L102
This pull request makes the interval to be configurable.

### CODE REVIEW

**Comment by moznion** [comment]
Sorry for late reaction. A pointed topic at the above is solved by 
https://github.com/soracom/lora-gateway-bridge/commit/293348a8dc485b5e33a6f17707b929df8e124950 💪

File: cmd/lora-gateway-bridge/cmd/configfile.go [hunk 1]
```diff
@@ -129,6 +129,9 @@ tls_cert="{{ .Backend.MQTT.TLSCert }}"
 
 # mqtt TLS key file (optional)
 tls_key="{{ .Backend.MQTT.TLSKey }}"
+
+# maximum seconds that will be waited between reconnection attempts when connection is lost
+max_reconnect_interval_seconds="{{ .Backend.MQTT.MaxReconnectIntervalSeconds }}"
```
Comment (brocaar) [inline cmd/lora-gateway-bridge/cmd/configfile.go]: Could you rename this to `max_reconnect_interval` and `MaxReconnectIntervalSeconds` to `MaxReconnectInterval`?

E.g.

```
# Max reconnect interval.
#
# The maximum interval that will be waited between reconnection attempts when the connection is lost.
+max_reconnect_interval="{{ .Backend.MQTT.MaxReconnectInterval }}"
```

File: cmd/lora-gateway-bridge/cmd/root.go [hunk 2]
```diff
@@ -79,6 +81,7 @@ func init() {
 	viper.SetDefault("backend.mqtt.config_topic_template", "gateway/{{ .MAC }}/config")
 	viper.SetDefault("backend.mqtt.server", "tcp://127.0.0.1:1883")
 	viper.SetDefault("backend.mqtt.clean_session", true)
+	viper.SetDefault("backend.mqtt.max_reconnect_interval_seconds", 10*time.Minute)
```
Comment (brocaar) [inline cmd/lora-gateway-bridge/cmd/root.go]: `max_reconnect_interval_seconds` -> `max_reconnect_interval`

File: docs/content/install/config.md [hunk 3]
```diff
@@ -192,6 +192,9 @@ tls_cert=""
 
 # mqtt TLS key file (optional)
 tls_key=""
+
+# maximum seconds that will be waited between reconnection attempts when connection is lost
+max_reconnect_interval_seconds=600
```
Comment (brocaar) [inline docs/content/install/config.md]: This then becomes `max_reconnect_interval="10m0s"`

```
# Max reconnect interval.
#
# The maximum interval that will be waited between reconnection attempts when the connection is lost.
max_reconnect_interval="10m0s"
```

File: internal/backend/mqttpubsub/backend.go [hunk 4]
```diff
@@ -19,21 +19,22 @@ import (
...
+	CACert                      string          `mapstructure:"ca_cert"`
+	TLSCert                     string          `mapstructure:"tls_cert"`
+	TLSKey                      string          `mapstructure:"tls_key"`
+	UplinkTopicTemplate         string          `mapstructure:"uplink_topic_template"`
+	DownlinkTopicTemplate       string          `mapstructure:"downlink_topic_template"`
+	StatsTopicTemplate          string          `mapstructure:"stats_topic_template"`
+	AckTopicTemplate            string          `mapstructure:"ack_topic_template"`
+	ConfigTopicTemplate         string          `mapstructure:"config_topic_template"`
+	AlwaysSubscribeMACs         []lorawan.EUI64 `mapstructure:"-"`
+	MaxReconnectIntervalSeconds time.Duration   `mapstructure:"max_reconnect_interval_seconds"`
```
Comment (brocaar) [inline internal/backend/mqttpubsub/backend.go]: This then becomes:

```go
MaxReconnectInterval time.Duration   `mapstructure:"max_reconnect_interval"`
```

File: internal/backend/mqttpubsub/backend.go [hunk 5]
```diff
@@ -100,6 +101,7 @@ func NewBackend(c BackendConfig) (*Backend, error) {
 	opts.SetClientID(b.config.ClientID)
 	opts.SetOnConnectHandler(b.onConnected)
 	opts.SetConnectionLostHandler(b.onConnectionLost)
+	opts.SetMaxReconnectInterval(b.config.MaxReconnectIntervalSeconds * time.Second)
```
Comment (brocaar) [inline internal/backend/mqttpubsub/backend.go]: And this becomes `opts.SetMaxReconnectInterval(b.config.MaxReconnectInterval)`

**Comment by brocaar** [comment]
> Sorry for late reaction. A pointed topic at the above is solved by
soracom/lora-gateway-bridge@293348a 💪

That link doesn't work for me? I've added some in-line comments. Could you make these changes? Alternatively, I can also pull your changes make the modifications and push these. Let me know what you prefer.

**Comment by moznion** [comment]
@brocaar 
I'm so sorry, I posted wrong URL. Actually https://github.com/moznion/lora-gateway-bridge/commit/293348a8dc485b5e33a6f17707b929df8e124950 is correct.

**Comment by brocaar** [comment]
Why did you change the `MaxReconnectInterval` from `time.Duration` to `string`? When you use `time.Duration` and set the default to `10 * time.Minute` you don't have to deal with the `string -> time.Duration` parsing yourself as this is done automatically for you :-) Could you change this back?

**Comment by moznion** [comment]
I didn't know that the `time.Duration` style string option is converted to `time.Duration` automatically.

Thank you for pointing it out. I fixed that at 192e164.

**Comment by brocaar** [comment]
Thanks @moznion 👍

**Pull request merged**